crates/but-core/src/cmd.rs

```diff
@@ -4,12 +4,21 @@ use std::ffi::OsString;
 ///
 /// On Windows, this specifically uses the Git-bundled shell, further increasing compatibility.
 pub fn prepare_with_shell(program: impl Into<OsString>) -> gix::command::Prepare {
+    let login_shell = std::env::var_os("SHELL");
+    // if login shell is set and it is a POSIX shell, use it, otherwise use the default shell
+    let shell_program = login_shell
+        .filter(|s| {
+            let s = s.to_string_lossy();
+            s.ends_with("/sh") || s.ends_with("/bash") || s.ends_with("/zsh")
```

Comment:
A great idea, which I hope will make signing work on more systems.

When hooks or worktree-filters come into play, I have a feeling this function is less useful as the environment of the parent process becomes relevant as the program launch isn't usually under our direct control.

Once this is sorted though, I think GB can launch things just like Git and it will work as if GB was run from a user's terminal.